### PR TITLE
[IOTDB-4498] Update StatusCode to the latest

### PR DESCRIPTION
**Pull request by qiaojialin**
https://apache-iotdb.feishu.cn/docx/KzecdhMLWo2jDqx2y0Xc0ELRnCd

1. Make error status end with ERROR
2. Remove some unused StatusCode
3. Adjust some names and messages.

The adjustment of code value will be made in the next pr.

### CODE REVIEW

**Review by HTHou — APPROVED** [review]

**Review by MarcosZyk — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java [hunk 1]
```diff
@@ -1188,13 +1189,13 @@ private TSStatus transferLeader(ConsensusGroupId regionId, Peer newLeaderPeer) {
     } else if (regionId instanceof SchemaRegionId) {
       resp = SchemaRegionConsensusImpl.getInstance().transferLeader(regionId, newLeaderPeer);
     } else {
-      status.setCode(TSStatusCode.REGION_LEADER_CHANGE_FAILED.getStatusCode());
+      status.setCode(TSStatusCode.REGION_LEADER_CHANGE_ERROR.getStatusCode());
```
Comment (Beyyes) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:1192]: do we put v. at the front? so it should be `CHANGE_REGION_LEADER_ERROR`?

File: docs/UserGuide/API/Status-Codes.md [hunk 2]
```diff
@@ -46,64 +46,122 @@ Here is a list of Status Code and related message:
...
+|712|DELETE_REGION_ERROR|Delete region error|
+|713|PARTITION_CACHE_UPDATE_ERROR|Update partition cache failed|
+|714|DESERIALIZE_PIECE_OF_TSFILE_ERROR|Error when deserialize a piece of TsFile|
+|715|CONSENSUS_NOT_INITIALIZED|Consensus is not initialized and cannot provide service|
+|800|CONFIGURATION_ERROR|Configuration error|
+|901|DATANODE_ALREADY_REGISTERED|DataNode already registered in cluster|
+|903|DATABASE_ALREADY_EXISTS|Database already exist|
+|904|NO_ENOUGH_DATANODE|The number of DataNode is not enough, cannot remove DataNode or create enough replication|
+|905|ERROR_GLOBAL_CONFIG|Global config in cluster does not consistent|
+|906|ADD_CONFIGNODE_ERROR|Add ConfigNode failed|
```
Comment (Beyyes) [inline docs/UserGuide/API/Status-Codes.md:149]:
```suggestion
|906|ADD_CONFIGNODE_ERROR|Add ConfigNode error|
```

File: docs/UserGuide/API/Status-Codes.md [hunk 3]
```diff
@@ -46,64 +46,122 @@ Here is a list of Status Code and related message:
...
+|907|REMOVE_CONFIGNODE_ERROR|Remove ConfigNode failed|
```
Comment (Beyyes) [inline docs/UserGuide/API/Status-Codes.md:150]:
```suggestion
|907|REMOVE_CONFIGNODE_ERROR|Remove ConfigNode error|
```

File: docs/UserGuide/API/Status-Codes.md [hunk 4]
```diff
@@ -46,64 +46,122 @@ Here is a list of Status Code and related message:
...
+|912|DATANODE_NOT_EXIST|DataNode not exist error|
```
Comment (Beyyes) [inline docs/UserGuide/API/Status-Codes.md:151]:
```suggestion
|912|DATANODE_NOT_EXIST_ERROR|DataNode not exist error|
```

File: docs/UserGuide/API/Status-Codes.md [hunk 5]
```diff
@@ -46,64 +46,122 @@ Here is a list of Status Code and related message:
...
+|917|DATANODE_STOP_FAILED|DataNode stop error|
```
Comment (Beyyes) [inline docs/UserGuide/API/Status-Codes.md:152]:
```suggestion
|917|DATANODE_STOP_ERROR|DataNode stop error|
```

**Review by Beyyes — APPROVED** [review]

File: docs/UserGuide/API/Status-Codes.md (hunk 2 above)
Comment (qiaojialin) [reply docs/UserGuide/API/Status-Codes.md:149]: fixed

File: docs/UserGuide/API/Status-Codes.md (hunk 5 above)
Comment (qiaojialin) [reply docs/UserGuide/API/Status-Codes.md:152]: fix

**Pull request merged**